src/Core/Router.php

```diff
@@ -40,12 +40,30 @@ public function patch($uri, $controllers)
     public function route($method, $uri)
     {
         foreach ($this->routes as $routes) {
-            if (strtoupper($method) === $routes['method'] && $uri === $routes['uri']) {
-                [$className , $methodName] = stringToArray("::" , $routes["controllers"]);
-                require base_path("Http/Controllers/{$className}.php");
-                $className = "\App\Http\Controllers\\" . $className;
-                (new $className)->$methodName();
+
+            if (strtoupper($method) === $routes['method']) {
+
+                $pattern = preg_replace('/\{([^\/]+)\}/', '([^/]+)', $routes['uri']);
+
+                $pattern = "#^" . $pattern . "$#";
+
+                if (preg_match($pattern, $uri, $matches)) {
+
+
+                    array_shift($matches);
+                    
+
+                    [$className, $methodName] = stringToArray("::", $routes["controllers"]);
+
+                    require base_path("Http/Controllers/{$className}.php");
```

Comment:
Check file existence for better error-handling and debugging. Something like this
```php
if (!file_exists($controllerPath)) { 
  abort(500, 'Controller not found'); 
}
```